db/error_handler.cc

```diff
@@ -389,16 +442,25 @@ Status ErrorHandler::ClearBGError() {
 Status ErrorHandler::RecoverFromBGError(bool is_manual) {
 #ifndef ROCKSDB_LITE
   InstrumentedMutexLock l(db_mutex_);
+  bool no_bg_work_original_flag = soft_error_no_bg_work_;
   if (is_manual) {
     // If its a manual recovery and there's a background recovery in progress
     // return busy status
     if (recovery_in_prog_) {
       return Status::Busy();
     }
     recovery_in_prog_ = true;
+
+    // In manual resume, we allow the bg work to run. If it is a auto resume,
+    // the bg work should follow this tag.
+    soft_error_no_bg_work_ = false;
+
+    // In manual resume, the flush reason is kErrorRecovery.
+    recover_context_.flush_reason = FlushReason::kErrorRecovery;
   }
 
-  if (bg_error_.severity() == Status::Severity::kSoftError) {
+  if (bg_error_.severity() == Status::Severity::kSoftError &&
+      no_bg_work_original_flag == false) {
```

Comment:
Can ```recover_context_.flush_reason``` be used here? It feels a little non-intuitive to rely on ```no_bg_work_original_flag```, since that's mainly to determine how other BG work should behave.